internal/cli/command/controlplane/debug.go

```diff
@@ -0,0 +1,241 @@
+	logResult("add pipeline/resources/banzaicloud/secrets.txt", addFile(tarWriter, "pipeline/resources/banzaicloud/secrets.txt", fetchContainerCommandOutputAndError(options.cpContext, []string{"kubectl", "get", "secrets", "-owide", "-nbanzaicloud"}, env)))          // -owide does not include contents
+	logResult("add pipeline/resources/banzaicloud/events.yaml", addFile(tarWriter, "pipeline/resources/banzaicloud/events.yaml", fetchContainerCommandOutputAndError(options.cpContext, []string{"kubectl", "get", "events", "-oyaml", "-nbanzaicloud"}, env)))
+	// TODO add helm binary to installer image
+	logResult("add pipeline/resources/banzaicloud/helm_list.txt", addFile(tarWriter, "pipeline/resources/banzaicloud/helm_list.txt", fetchContainerCommandOutputAndError(options.cpContext, []string{"helm", "list", "--namespace", "banzaicloud", "--all"}, env)))
+
+	logResult("create pipeline/logs", addDir(tarWriter, "pipeline/logs"))
+	logResult("create pipeline/logs/banzaicloud", addDir(tarWriter, "pipeline/logs/banzaicloud"))
+	pods := strings.Split(strings.TrimSpace(fetchContainerCommandOutputAndError(options.cpContext, []string{"kubectl", "get", "pods", "-oname", "-nbanzaicloud"}, env)), "\n")
+	for _, pod := range pods {
+		pod = strings.TrimPrefix(strings.TrimSpace(pod), "pod/")
+		logResult("add pod log", addFile(tarWriter, filepath.Join("pipeline/logs/banzaicloud/", pod+".log"), fetchContainerCommandOutputAndError(options.cpContext, []string{"kubectl", "logs", "-nbanzaicloud", pod, "--all-containers"}, env)))
+	}
+
+	log.Infof("debug bundle has been written to %q", path)
+	logResult("add meta.log", addFile(tarWriter, "meta.log", logBuffer.String()))
+
+	return nil
+}
+
+// simpleCommand runs the given shell command, and returns its outputs, an error message or both
+func simpleCommand(command string, args ...string) string {
+	ctx, cancel := context.WithTimeout(context.Background(), 60*time.Second)
+	defer cancel()
+
+	cmd := exec.CommandContext(ctx, "sh", "-c", command)
+	if len(args) > 0 {
+		cmd = exec.CommandContext(ctx, command, args...)
+	}
+	out, err := cmd.CombinedOutput()
+	output := strings.TrimSpace(string(out))
+	if err != nil {
+		if output != "" {
+			output += "\n"
+		}
+		output += err.Error()
+	}
+	return output
+}
+
+func logResult(what string, err error) {
+	if err == nil {
+		log.Debugf("%s succeeded", what)
+	} else {
+		log.Errorf("%s failed: %v", what, err)
+	}
+}
+
+func addDir(w *tar.Writer, name string) error {
+	name = strings.TrimRight(filepath.Join(basePath, name), "/") + "/"
+	err := w.WriteHeader(&tar.Header{Name: name, Mode: 0777})
+	return errors.WrapIf(err, "failed to create directory in archive")
+}
+
+func addFile(w *tar.Writer, name, content string) (err error) {
+	name = filepath.Join(basePath, name)
+	bytes := []byte(content)
+	err = w.WriteHeader(&tar.Header{Name: name, Mode: 0666, Size: int64(len(bytes))})
```

Comment:
**Recommended**: i think this will trigger [GoSec G302: Poor file permissions used with chmod](https://github.com/securego/gosec/blob/master/rules/fileperms.go#L82), we **SHOULD** use `0o600` in case the tar extracting user becomes the owner at the recipient's system and `0o244` in case the original tar creator user becomes the owner.

Reply:
if you extract this, tar will respect *your* umask value and of course it'll extract the files with your ownership unless running tar as root (which coincidentally would end up in the same result, because we create the archive with uid=0 files)

Reply:
Then I suggest using a `0o600` permission value here.

Reply:
I don't understand why. The convention for command line tools is to create files with mode 666 or 777 (which is applicable) unless a specific reason exists, and let the umask decide the final permissions of the files. This is exactly what all the core tools (like `touch` or `mkdir`) or the shell's io redirection options do, so a user shouldn't ever be surprised. It's quite uncommon to have a default umask less than 022, but for example if you want extracted files to be writeable by group members, you still have the control to set umask to 002.

Reply:
I think it is a generalization to say the convention for command line tools is to use `666` and `777` permissions in all cases. I think this should be use-case specific. For example if a CLI password manager writes a vault file (even if encrypted), it sure as hell shouldn't write it with `777`.

If we are arguing whether the current use-case warrants for a narrower permission set I didn't see why `600` wouldn't be enough (AFAIK at the extraction the extractor would be able to access it anyway), which would be optimal from security and linting perspective as well, but I can let this go with the comment we will need a nolint comment for it once gosec is turned on.

I merely commented this on the review because I foresaw a probable linting problem and thought it can be easily fixed with a narrower permission set which seemed sufficient for the use-case as well.

If I made a mistake somewhere, feel free to point it out to me so I can adjust my learning model accordingly.

Reply:
I let this be postponed until/if we use said linter.